File: backend/SNS.Application/Projects/Contracts/ProjectDetailsDto.cs
using SNS.Domain.Projects.Enums;

namespace SNS.Application.Projects.Contracts;

public record ProjectDetailsDto(
    Guid ProjectId,
    string Title,
    string ShortDescription,
    string MainImageUrl,
    string ReadmeContent,
    string GitHubUrl,
    string LiveDemoUrl,
    ProjectType Type,
    ProjectStatus Status,
    bool IsActive,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    List<ProjectSkillDto> Skills,
    List<ProjectTagDto> Tags,
    int SaveCount,
    int ViewCount
);

public record ProjectSkillDto(Guid SkillId, string SkillName);
public record ProjectTagDto(Guid TagId, string TagName);

File: SNS.Web/src/app/projects/contracts/project-details.dto.ts
import { ProjectType } from '../enums/project-type.enum';
import { ProjectStatus } from '../enums/project-status.enum';
import { ProjectSkillDto } from './project-skill.dto';
import { ProjectTagDto } from './project-tag.dto';

export interface ProjectDetailsDto {
    projectId: string;
    ownerId: string;
    id?: string;
    title: string;
    shortDescription: string;
    mainImageUrl: string;
    readmeContent: string;
    gitHubUrl: string;
    liveDemoUrl: string;
    type: ProjectType;
    status: ProjectStatus;
    isActive: boolean;
    createdAt: string | Date;
    updatedAt: string | Date;
    skills: ProjectSkillDto[];
    tags: ProjectTagDto[];
    saveCount: number;
    viewCount: number;
}
